### PR TITLE
[SDL 0253] New vehicle data StabilityControlsStatus

### LINKED ISSUE #253: [SDL 0253] New vehicle data StabilityControlsStatus
Proposal: [New vehicle data StabilityControlsStatus](https://github.com/smartdevicelink/sdl_evolution/edit/master/proposals/0253-New-vehicle-data-StabilityControlsStatus.md)

>This proposal is to add new vehicle data `StabilityControlsStatus` to know if stability controls like ESC, Traction Control etc. are active.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/841

### PR DESCRIPTION
**Pull request by MSavitsk**
Fixes #253

This PR is **[ready]** for review.

### Risk
This PR makes **[minor]** API changes.

### Testing Plan
Covered by existing unit tests.

### Summary
Applied [[SDL 0253] New vehicle data StabilityControlsStatus](https://github.com/smartdevicelink/sdl_evolution/edit/master/proposals/0253-New-vehicle-data-StabilityControlsStatus.md) changes

##### Enhancements
* Added required getters/setters in related RPC classes

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform) and this pull request adheres to the [Contributing Guide](./CONTRIBUTING.md).

### CODE REVIEW

**Comment by MSavitsk** [comment]
@kshala-ford @mrapitis please review the changes.

**Comment by santhanamk** [comment]
Hi @MSavitsk I am reviewing your PRs and am looking at the sdl javascript codebase for the first time.  I usually work with the sdl android codebase.

Your code looks good.  I have a suggestion:

1) Can you write some unit tests for `StabilityControlStatus`?  I think they would go under tests/node/rpc/structs

**Comment by vladmu** [comment]
@santhanamk thank you for review.
> 1. Can you write some unit tests for `StabilityControlStatus`?  I think they would go under tests/node/rpc/structs
> 2. Does `Test.js` need the new struct `StabilityControlsStatus`?

I will check the existing tests and add the required or answer here a little bit later.

**Comment by santhanamk** [comment]
Sounds good @vladmu.  Please let me know when you want me to review the code again.

**Comment by vladmu** [comment]
@santhanamk We've added unit tests, please review.

File: tests/Validator.js [hunk 1]
```diff
@@ -530,6 +530,23 @@ class Validator {
...
+     * @param {VehicleDataResult} item1 - A VehicleDataResult.
+     * @param {VehicleDataResult} item2 - A VehicleDataResult.
+     * @returns {Boolean} - Whether or not they're equal.
+     */
+    static validateVehicleDataResult (item1, item2) {
+        if (item1 === null || item2 === null) {
+            expect(item1).to.equal(item2);
+            return true;
+        }
+        expect(item1.getDataType()).to.be.equal(item2.getDataType());
```
Comment (santhanamk) [inline tests/Validator.js:545]: When comparing the two items' datatype, oemcustomdatatype, and resultcode does a null check need to be done to ensure that the datatype, oemcustomdatatype, or resultcode is not null?

Comment (vladmu) [reply tests/Validator.js:545]: this is done the same way as other validators, eg. `validateTemplateColorScheme` in this file.

Comment (santhanamk) [reply tests/Validator.js:545]: ok sounds good.  Thanks.

**Comment by santhanamk** [comment]
@theresalech this PR is ready for Livio review.

**Comment by crokita** [comment]
While we continue our review of this PR, please pull in the new changes from the develop branch. We are removing the built files from the project so that merge conflicts will stop happening with them and the processing of contributing and reviewing code will be easier.
@MSavitsk

**Comment by vladmu** [comment]
@crokita the branch is refreshed with new changes from develop. Please review.

**Pull request merged**